src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/components/SceneFunctionDetailsPanel/components/CodeContainer/domain/buildLineProfiles.ts

```diff
@@ -4,45 +4,39 @@ const VERTICAL_LINES_PADDING = 5;
 
 type CallSitesMap = Map<number, CallSiteProps>;
 
-export function buildPlaceholderLineProfiles(callSitesMap: CallSitesMap) {
+export function annotatePlaceholderLineProfiles(callSitesMap: CallSitesMap): LineProfile[][] {
   if (!callSitesMap.size) {
-    return [];
+    return [[], []];
   }
 
   const callSites = Array.from(callSitesMap.values()).sort((a, b) => a.line - b.line);
 
   const firstLineIndex = Math.max(0, callSites[0].line - VERTICAL_LINES_PADDING - 1);
   const lastLineIndex = callSites[callSites.length - 1].line + VERTICAL_LINES_PADDING + 1;
 
-  const lines = [];
-
+  const annotatedSnippet = [];
   for (let lineNumber = firstLineIndex + 1; lineNumber < lastLineIndex; lineNumber++) {
     const callSite = callSitesMap.get(lineNumber);
 
-    lines.push({
+    annotatedSnippet.push({
       line: undefined,
       number: lineNumber,
       cum: callSite?.cum ?? 0,
       flat: callSite?.flat ?? 0,
     });
   }
 
-  return lines;
+  // With no file contents, we return only a dummy annotated snippet which shows
+  // the appropriate line numbers, but no content.
+  return [annotatedSnippet, []];
 }
 
-export function buildLineProfiles(fileContent: string, callSitesMap: CallSitesMap): LineProfile[] {
-  if (!callSitesMap.size) {
-    return [];
-  }
-
+export function annotateLines(fileContent: string, callSitesMap: CallSitesMap): LineProfile[][] {
```

Comment:
Could we return a struct/map instead of nested array for better readability? e.g.

```
{ 
  snippetLines: LineProfile[],
  allLines: LineProfile[] 
}
```

instead of `LineProfile[][]`

Reply:
👍 fixed in https://github.com/grafana/explore-profiles/pull/459/commits/071a438d6955ee5c20092150789d20c5bb8a26f9.